File: 0601_0700/0603/0603.py
def eq(s, n, p, t, m):
    j = 0;
    for i in range(m):
        if p + j >= n:
            return False
        if t[i].isspace() and s[p + j].isspace():
            while p + j + 1 < n and s[p + j + 1].isspace():
                j += 1
        elif t[i] != s[p + j].lower():
            return False
        j += 1
    return True

with open("input.txt", "r") as f, open("output.txt", "w") as q:
    t = f.readline().strip().lower()
    s = [x for x in f.read()]
    m, n = len(t), len(s)
    p = 0
    while p < n - m:
        if eq(s, n, p, t, m):
            s.insert(p, "@")
            n += 1
            p += 1
        p += 1
        while p < n and s[p].isspace():
            p += 1
    q.write(''.join(s))

File: 0601_0700/0603/0603.c
#include <stdio.h>
#include <ctype.h>

int eq(char *s, char *t)
{
	for (int i = 0, j = 0; t[i]; ++i, ++j)
		if (isspace(t[i]) && (isspace(s[j])))
			while(isspace(s[j + 1]))
				++j;
		else if (t[i] != tolower(s[j]))
			return 0;
	return 1;
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int c, n = 0;
	char s[2001], t[101], *p = s;

	while((c = getc(f)) != '\n')
		t[n++] = tolower(c);
	t[n] = '\0';
	for (n = 0; (s[n] = getc(f)) != EOF; ++n);
	s[n] = '\0';
	while(*p)
	{
		if (eq(p, t))
			putc('@', q);
		putc(*p++, q);
		while (isspace(*p))
			putc(*p++, q);
	}
	return 0;
}
